src/core/core.datasetController.js

```diff
@@ -257,12 +296,14 @@ helpers.extend(DatasetController.prototype, {
 	},
 
 	linkScales: function() {
-		var chart = this.chart;
-		var meta = this._cachedMeta;
-		var dataset = this.getDataset();
-
-		meta.xAxisID = dataset.xAxisID || getFirstScaleId(chart, 'x');
-		meta.yAxisID = dataset.yAxisID || getFirstScaleId(chart, 'y');
+		const me = this;
+		const chart = me.chart;
+		const meta = me._cachedMeta;
+		const dataset = me.getDataset();
+		const xid = meta.xAxisID = dataset.xAxisID || getFirstScaleId(chart, 'x');
+		const yid = meta.yAxisID = dataset.yAxisID || getFirstScaleId(chart, 'y');
+		meta.xScale = me.getScaleForId(xid);
```

Comment:
this is a really good idea!

Reply:
do you think we should store `iScale` and `vScale` here as well?

Reply:
Probably